### DISCUSSION TITLE
Why doesn't the portal editor load the current revision?

### DISCUSSION THREAD

**Discussion by daninthemix** [Azure/api-management-developer-portal#2621] [Feature request]
As a test, I reverted the portal to default. And now whenever I launch the editor, it's permanently default. Doesn't matter if the portal is currently using a revision that has been customised. The portal editor only seems to load the most recent revision, even if that isn't the revision in use.

This is bone-headed design.

**Comment by malincrist** [comment]
hello @daninthemix , thanks for reaching out.
You are right, changing the portal revision only changes the "published" portal, not the designer mode.

The purpose behind this is to enable users to modify their portal while ensuring there's a well-functioning published version accessible to their customers. To illustrate, imagine this scenario: you're working on personalizing your developer portal and eventually publish it. Later on, you discover certain problems arising from these customizations that negatively impact the portal's usability. In such cases, you can promptly restore the published portal to a previous version that functions properly, all the while retaining the latest modifications in the designer interface to address the issues.

The solution would be to have a way to revert the designer content as well (separately from the published content) but unfortunately, we do not have this in plan yet. Thank you for understanding.

**Comment by daninthemix** [comment]
@malincrist thanks for the feature request. My issue with this is it if you publish a bad portal, it's literally impossible to revert without doing an APIM restore or fixing it manually.
